### ISSUE TITLE
Added weights have no impact on results

### ISSUE THREAD

**Issue by georg-at-github** [dmlc/mxnet#5392]
Hello,

I am trying to add weights to my NN's outputs before forwarding them into the SoftmaxOutput layer. The weights are fixed parameter that I pass to the module during binding. Strangely enough, these weights have no effect whatsoever on the softmax ouput:

``weights = mx.sym.Variable('weights')``
``...[various layers]...``
``fc2 = mx.sym.FullyConnected(...)``
``weighted_fc2 = fc2 + weights            ###This layer has no effect``
``loss = mx.sym.SoftmaxOutput(weighted_fc2, name='softmax')``

I have tried this with and without various weights and always obtained exactly the same outputs for all training steps (with given random seed). The module is created like this:

``weights = mx.nd.array(...)``
``self.module = mx.mod.Module(self.model, data_names=['weights', ...], fixed_param_names=['weights'])``
``self.module.bind( label_shapes=..., data_shapes=[('data',...), ..., ('weights', weights.shape)])``
``self.module.init_params(arg_params={..., 'weights': weights}, allow_missing=True)``

I am using Python 3.6 on Windows, MXNet last updated on 09/03/17.

Could you give me some advice on how to achieve what I am trying to do, namely adding a fixed quantity to the output before sending it into the SoftmaxOutput?

Many thanks in advance for any clarifications!

**Comment by georg-at-github** [comment]
Through a monitor I found out that 'weights' is always initialized to zero! This is why adding the weights has no effect. I open a new issue for this one.
